### PR TITLE
Apply iOS status bar padding only to fullscreen views

### PR DESCRIPTION
**Pull request by cbracken**
Previously padding was applied to account for the status bar, whether in
standard or expanded 'in-call' geometry only if the current view was not
a subview of a containing view. This didn't cover the case of root views
embedded in other windows (e.g. dialogs).

We also ensure that the window is fullscreen to account for cases like
small dialogs centred on the screen.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -341,23 +341,33 @@ - (void)updateViewportMetrics {
...
+  if (![self isWindowFullscreen]) {
+    return 0.0;
+  }
+
+  UIScreen *screen = self.view.window.screen;
+  CGRect statusFrame = [UIApplication sharedApplication].statusBarFrame;
+  CGRect viewFrame = [self.view convertRect:self.view.bounds
+                          toCoordinateSpace:screen.coordinateSpace];
+  CGFloat padding = statusFrame.origin.y + statusFrame.size.height - viewFrame.origin.y;
+  return MAX(padding, 0.0);
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: This should never be negative assuming the status frame has origin 0,0... but still.

**Comment by cbracken** [comment]
@chinmaygarde

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm (hunk 1 above)
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Could probably avoid the `MAX()` by doing a `CGRectIntersectsRect()` check, though this seems cheaper.

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 2]
```diff
@@ -341,23 +341,38 @@ - (void)updateViewportMetrics {
...
+- (bool)isWindowFullscreen {
+  UIWindow *window = self.view.window;
+  return CGRectEqualToRect(window.frame, window.screen.bounds);
+}
+
+- (CGFloat)statusBarPadding {
+  // If we're a child of a containing view, let the container apply padding.
+  if (self.parentViewController != nil) {
+    return 0.0;
+  }
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: Any case where this isn't a legitimate assumption?

Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: > Any case where this isn't a legitimate assumption?

I am uncertain. I'd keep this check in place. In the future, if embedders want to have more control over this behavior, we can add a `FlutterViewControllerDelegate` and ask it the embedder is handling status bar behavior or us. I feel like we have already covered all our bases with this patch.

**Pull request merged**